### PR TITLE
weapon_class

### PR DESCRIPTION
**Pull request by Rolandevitt**
Changed class Weapon, added class Fighter(Creature). World.tick is endless now..

### CODE REVIEW

File: creature.py [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
     def __init__(self):
-        self.health = 50
+        self.health = 25
         self.name = f"creature_{Creature.id}"
+        self.ac = 11
         self.exp = 0
         self.lvl = 1
+        self.weapon = weapons[random.randint(0, (len(weapons) - 1))]
+        self.dmg = self.weapon.damage()
+        self.what_weapon = self.weapon.name
```
Comment (678098) [inline creature.py:16]: Bad name for a field. Field should have name with some statement without question.

File: creature.py [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+
 
 class Creature:
     id = 1
 
     def __init__(self):
-        self.health = 50
+        self.health = 25
         self.name = f"creature_{Creature.id}"
+        self.ac = 11
```
Comment (678098) [inline creature.py:11]: What is `ac`? It's not good to give such names so it's not obvious what it means.

File: creature.py (hunk 1 above)
Comment (678098) [reply creature.py:16]: Either `weapon_name` should work or you just re-use `self.weapon.name` field where it's needed. Why duplicate information? What if you change the `weapon` but forget to change `self.weapon_name`? You can also define property `weapon_name` if it's necessary.

File: creature.py [hunk 3]
```diff
@@ -18,20 +24,60 @@ def creatures_created(cls) -> int:
...
+
+    def __repr__(self) -> str:
+        return f"\nCreature [name: {self.name}, health: {self.health}, lvl: {self.lvl}, exp: {self.exp}, weapon: {self.weapon}]"
+
+
+class Monsters(Creature):
+    id = 1
+    def __init__(self):
+        super().__init__()
+        self.name = f"monster_{Monsters.id}"
```
Comment (678098) [inline creature.py:56]: You don't have to re-use this pattern for naming just everywhere.

File: creature.py [hunk 4]
```diff
@@ -18,20 +24,60 @@ def creatures_created(cls) -> int:
...
+
+
+class Fighters(Creature):
+    id = 1
+    def __init__(self):
+        super().__init__()
+        self.name = f"fighter_{Fighters.id}"
+        self.health = random.randint(5, 15)  # hit points
+        self.dex = random.randint(1, 3) #dexterity
+        self.ac = armors[random.randint(0, 3)].base_ac + self.dex  # armor class
```
Comment (678098) [inline creature.py:74]: These comments are useless, so these should be removed. It should be obvious from the fields's name what it is, `health` or `dexterity`.

File: creature.py [hunk 5]
```diff
@@ -18,20 +24,60 @@ def creatures_created(cls) -> int:
...
 
     def __repr__(self) -> str:
-        return f"Creature [name: {self.name}, health: {self.health}, lvl: {self.lvl}, exp: {self.exp}]"
\ No newline at end of file
+        return f"\nFighter [name: {self.name}, health: {self.health}, lvl: {self.lvl}, exp: {self.exp}, weapon: {self.what_weapon} with damage: {self.weapon.num_rolls}d{self.weapon.dice}]"
+
+
+if __name__ == "__main__":
+    print([Monsters() for _ in range (5)])
+    print([Fighters() for _ in range (5)])
```
Comment (678098) [inline creature.py:83]: Strange backspace after `range`. You can use auto-formatter `Ctrl + Alt + L`

File: weapon.py [hunk 6]
```diff
@@ -2,23 +2,47 @@
...
-    def __init__(self, damage, name, rarity, cost):
-        self.damage = damage
+    def __init__(self, dice, num_rolls, name, rarity, cost):
+        self.dice = dice
+        self.num_rolls = num_rolls
         self.name = name
         self.rarity = rarity
         self.cost = cost
 
+    #@property
```
Comment (678098) [inline weapon.py:12]: This comment should be removed

File: weapon.py [hunk 7]
```diff
@@ -2,23 +2,47 @@
...
-    Weapon(Dice(1, 6).show_dice(), "Long sword", 0.5, 100),
-    Weapon(Dice(2, 6).show_dice(), "Wolf Knight's Greatsword", 0.8, 800),
-    Weapon(Dice(1, 12).show_dice(), "Uchigatana", 0.6, 500),
-    Weapon(Dice(2, 4).show_dice(), "Sellsword Twinblades", 0.5, 200),
-    Weapon(Dice(1, 4).show_dice(), "Dagger", 0.4, 50)
+    Weapon(Dice(6), 1, "Long sword", 0.5, 100),
+    Weapon(Dice(6), 2, "Wolf Knight's Greatsword", 0.8, 800),
+    Weapon(Dice(12), 1, "Uchigatana", 0.6, 500),
+    Weapon(Dice(4), 2, "Sellsword Twinblades", 0.5, 200),
+    Weapon(Dice(4), 1, "Dagger", 0.4, 50)
```
Comment (678098) [inline weapon.py:24]: Learn how to declare data in json and load weapon parameters from json-configuration file.

File: creature.py [hunk 8]
```diff
@@ -18,20 +24,60 @@ def creatures_created(cls) -> int:
...
+        Monsters.id += 1
+
+    def lvl_up(self):
+        pass
+
+    def __repr__(self) -> str:
+        return f"\nMonster [name: {self.name}, health: {self.health}, weapon: {self.weapon}]"
+
+
+class Fighters(Creature):
```
Comment (678098) [inline creature.py:67]: It's very strange to name single `Fighter` or `Monster` with `s`, should be just `Fighter` or `Monster`.

File: weapon.py [hunk 9]
```diff
@@ -2,23 +2,47 @@
...
 ]
 
-print(weapons)
+
+class Armor():
```
Comment (678098) [inline weapon.py:28]: Better remove empty `()` in a class declaration, it's confusing.

File: world.py [hunk 10]
```diff
@@ -8,23 +9,39 @@ def __init__(self):
...
     def run(self):
-        while len(self.creatures) > 1:
-            self.tick()
\ No newline at end of file
+        while len(self.monsters) > 0 and len(self.fighters) > 0:
+            self.tick()
+        if len(self.monsters) == 0:
+            print("Fighters win!")
+        else:
+            print("Monsters win!")
```
Comment (678098) [inline world.py:47]: Empty line in the end of a file

**Review by 678098 — CHANGES_REQUESTED** [review]

File: weapon.py (hunk 9 above)
Comment (Rolandevitt) [reply weapon.py:28]: Ok

**Pull request merged**